storm-server/src/main/java/org/apache/storm/daemon/nimbus/Nimbus.java

```diff
@@ -3165,6 +3163,16 @@ public void submitTopologyWithOpts(String topoName, String uploadedJarLocation,
                 topoConf.remove(Config.TOPOLOGY_CLASSPATH_BEGINNING);
             }
 
+            // storm.messaging.netty.authentication is about inter-worker communication
+            // enforce netty authentication when either topo or daemon set it to true
+            boolean enforceNettyAuth = (Boolean) topoConf.getOrDefault(Config.STORM_MESSAGING_NETTY_AUTHENTICATION, false)
```

Comment:
`storm.messaging.netty.authentication` is guaranteed to be set because it is in `defaults.yaml`. We don't want to have default value in different places.